Fails to build on hurd-i386 and GNU/kFreeBSD

Previous versions of syslog-ng built fine on the Hurd and GNU/kFreeBSD, 3.6.1 (and later) do not, due to the credential stuff in afsocket:

```
libtool: compile:  gcc -std=gnu99 -DHAVE_CONFIG_H -I. -I../.. -I../../lib -I../../modules -I./lib -I./modules -I../../modules/afsocket -I./modules/afsocket -D_FORTIFY_SOURCE=2 -pthread -I/usr/include/glib-2.0 -I/usr/lib/i386-gnu/glib-2.0/include -I/usr/include/eventlog -DLIBNET_LIL_ENDIAN -I/usr/include/dbi -I/usr/include/dbi/dbi -D_GNU_SOURCE -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -Wall -pthread -c ../../modules/afsocket/transport-mapper-inet.c  -fPIC -DPIC -o modules/afsocket/.libs/modules_afsocket_libafsocket_notls_la-transport-mapper-inet.o
../../modules/afsocket/transport-unix-socket.c:148:48: error: unknown type name 'cred_t'
 _feed_aux_from_ucred(LogTransportAuxData *aux, cred_t *uc)
                                                ^
../../modules/afsocket/transport-unix-socket.c: In function 'log_transport_unix_socket_new':
../../modules/afsocket/transport-unix-socket.c:251:3: warning: implicit declaration of function 'socket_set_pass_credentials' [-Wimplicit-function-declaration]
   socket_set_pass_credentials(fd);
   ^
../../modules/afsocket/transport-unix-socket.c: At top level:
../../modules/afsocket/transport-unix-socket.c:36:1: warning: '_add_nv_pair_int' defined but not used [-Wunused-function]
 _add_nv_pair_int(LogTransportAuxData *aux, const gchar *name, gint value)
 ^
../../modules/afsocket/transport-unix-socket.c:102:1: warning: '_add_nv_pair_proc_read_unless_unset' defined but not used [-Wunused-function]
 _add_nv_pair_proc_read_unless_unset(LogTransportAuxData *aux, const gchar *name, pid_t pid, const gchar *proc_file, const gchar *unset_value)
 ^
../../modules/afsocket/transport-unix-socket.c:115:1: warning: '_add_nv_pair_proc_read_argv' defined but not used [-Wunused-function]
 _add_nv_pair_proc_read_argv(LogTransportAuxData *aux, const gchar *name, pid_t pid, const gchar *proc_file)
 ^
../../modules/afsocket/transport-unix-socket.c:135:1: warning: '_add_nv_pair_proc_readlink' defined but not used [-Wunused-function]
 _add_nv_pair_proc_readlink(LogTransportAuxData *aux, const gchar *name, pid_t pid, const gchar *proc_file)
 ^
make[4]: *** [modules/afsocket/modules_afsocket_libafsocket_notls_la-transport-unix-socket.lo] Error 1
```

Full build log available [here (for hurd)](https://buildd.debian.org/status/fetch.php?pkg=syslog-ng&arch=hurd-i386&ver=3.6.1-1&stamp=1415742821). and [here (for kfreebsd)](https://buildd.debian.org/status/fetch.php?pkg=syslog-ng&arch=kfreebsd-amd64&ver=3.6.1-1&stamp=1415735941)

--- a/modules/afsocket/transport-unix-socket.c
+++ b/modules/afsocket/transport-unix-socket.c
@@ -144,6 +144,7 @@ _add_nv_pair_proc_readlink(LogTransportAuxData *aux, const gchar *name, pid_t pi
     log_transport_aux_data_add_nv_pair(aux, name, content);
 }
 
+#if defined (CRED_PASS_SUPPORTED)
 static void
 _feed_aux_from_ucred(LogTransportAuxData *aux, cred_t *uc)
 {
@@ -151,6 +152,7 @@ _feed_aux_from_ucred(LogTransportAuxData *aux, cred_t *uc)
   _add_nv_pair_int(aux, ".unix.uid", cred_get(uc, uid));
   _add_nv_pair_int(aux, ".unix.gid", cred_get(uc, gid));
 }
+#endif
 
 #if defined(__linux__) && defined(CRED_PASS_SUPPORTED)
 static void

